### ISSUE TITLE
Remove the hard dependency on a running bitcoind

### ISSUE THREAD

**Issue by darosior** [wizardsardine/liana#883]
I originally implemented `lianad` with a hard dependency on a running `bitcoind`: it must be present at startup and if it becomes unreachable for more than X minutes we crash. It made it easier to reason about operations as we could assume it was always present. However there are some advantages in removing this hard dependency:
- Get rid of all the assertions in the bitcoind communication code. These can just become soft errors, which can be logged and/or passed on to the user interface (RPC or the GUI). This improves the UX (as long as we make it clear that the information shown aren't up to date).
- The user would still be able to access the wallet during IBD. It doesn't appear to be much, but currently the mandatory IBD progress bar before being able to access the wallet really hurts UX.
- If the first startup is actually a user willing to import an existing descriptor, today they would have to wait for IBD and *then* rescan whereas this makes it possible to load the descriptor and then IBD (removing the need for an additional rescan).
- First step toward not relying on `bitcoind` for transaction storage (and instead use our own DB) which is necessary to introduce different Bitcoin backends.

Cons:
- Confusion that most features wouldn't work.
- Confusion that information displayed may be out of date. While on the GUI a global warning could be displayed, on the RPC interface i don't think it's reasonable to add one to each command.
- Related to the previous point, this shifts the burden of checking we are actually connected to the Bitcoin network from us to the client (the GUI or the consumer of the RPC API).

Any thoughts?

**Comment by pythcoiner** [comment]
some kind related to #672

**Comment by darosior** [comment]
Yes although we would still need `bitcoind`. Being able to start without it would only be a hack.

**Comment by pythcoiner** [comment]
I'm in favor of this as it give ability to use Liana in some kind of `offline mode`

**Comment by pythcoiner** [comment]
as an example sparrow uses a a toggle button to turn on/off to the bitcoin backend:
![image](https://github.com/wizardsardine/liana/assets/124568858/fc1f172c-0f52-4df3-9caa-9c1e1c82baad)

**Comment by darosior** [comment]
> The user would still be able to access the wallet during IBD. It doesn't appear to be much, but currently the mandatory IBD progress bar before being able to access the wallet really hurts UX.

Actually this is unrelated, this can already be done today right @edouardparis? It's only the poller which doesn't start polling until IBD is done.

**Comment by darosior** [comment]
> If the first startup is actually a user willing to import an existing descriptor, today they would have to wait for IBD and then rescan whereas this makes it possible to load the descriptor and then IBD (removing the need for an additional rescan).

Same here, it's actually what we do already!

**Comment by darosior** [comment]
> First step toward not relying on bitcoind for transaction storage (and instead use our own DB) which is necessary to introduce different Bitcoin backends.

Even for this i'm not sure. We can do this without making a running Bitcoin backend a hard dependency of the software.
